### PR TITLE
Revert "Merge pull request #5323 from scivision/pgiflang"

### PR DESCRIPTION
**Pull request by jpakkane**
This reverts commit 9cd89f55fca0933dd2897ac7470c5827d939e0ac, reversing
changes made to 60969d99d33001317c569a9b37d3b9efae08d387.

### CODE REVIEW

File: mesonbuild/environment.py [hunk 1]
```diff
@@ -1070,8 +1039,7 @@ def detect_d_compiler(self, want_cross):
         # up to date language version at time (2016).
         if exelist is not None:
             if os.path.basename(exelist[-1]).startswith(('ldmd', 'gdmd')):
-                raise EnvironmentException('Meson does not support {} as it is only a DMD frontend for another compiler.'
-                                           'Please provide a valid value for DC or unset it so that Meson can resolve the compiler by itself.'.format(exelist[-1]))
+                    raise EnvironmentException('Meson doesn\'t support %s as it\'s only a DMD frontend for another compiler. Please provide a valid value for DC or unset it so that Meson can resolve the compiler by itself.' % exelist[-1])
```
Comment (ignatenkobrain) [inline mesonbuild/environment.py]: __[Flake8]__ [E117] over-indented ([link](https://sider.review/gh/repos/19784232/pulls/5330#issue-4009639))

<sub>posted by Sider</sub>

File: mesonbuild/environment.py [hunk 2]
```diff
@@ -12,13 +12,14 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import os, platform, re, sys, shlex, shutil, subprocess
+import configparser, os, platform, re, sys, shlex, shutil, subprocess
```
Comment (ignatenkobrain) [inline mesonbuild/environment.py]: __[Flake8]__ [F401] 'configparser' imported but unused ([link](https://sider.review/gh/repos/19784232/pulls/5330#issue-4009637))

<sub>posted by Sider</sub>

File: mesonbuild/environment.py [hunk 3]
```diff
@@ -12,13 +12,14 @@
...
+import typing
```
Comment (ignatenkobrain) [inline mesonbuild/environment.py]: __[Flake8]__ [F401] 'typing' imported but unused ([link](https://sider.review/gh/repos/19784232/pulls/5330#issue-4009638))

<sub>posted by Sider</sub>

**Comment by dcbaker** [comment]
I don't have access to the PGI compilers (and I can't agree to the trial terms for reasons), if @scivision has tested the PGI compilers on windows and they work them I'm happy to leave that bit in and just revert the ICL patches. Honestly as long as we don't ship broken ICL support I'm fine leaving the patches in master as-is. I just don't want to have broken support in 0.51

**Comment by scivision** [comment]
What is the deadline for 0.51? I'm setup with latest releases of:

* Linux: Intel, PGI, flang, GCC
* Windows: Intel (deferred for now), PGI, GCC

**Comment by dcbaker** [comment]
I haven't heard a deadline yet, but I'm aiming to get ICL landed for 0.51 if at all possible.

**Comment by QuLogic** [comment]
Minor releases are generally on a monthly schedule.

File: mesonbuild/environment.py [hunk 4]
```diff
@@ -1073,8 +1042,7 @@ def detect_d_compiler(self, want_cross):
         # up to date language version at time (2016).
         if exelist is not None:
             if os.path.basename(exelist[-1]).startswith(('ldmd', 'gdmd')):
-                raise EnvironmentException('Meson does not support {} as it is only a DMD frontend for another compiler.'
-                                           'Please provide a valid value for DC or unset it so that Meson can resolve the compiler by itself.'.format(exelist[-1]))
+                    raise EnvironmentException('Meson doesn\'t support %s as it\'s only a DMD frontend for another compiler. Please provide a valid value for DC or unset it so that Meson can resolve the compiler by itself.' % exelist[-1])
```
Comment (ignatenkobrain) [inline mesonbuild/environment.py]: __[Flake8]__ [E117] over-indented ([link](https://sider.review/gh/repos/19784232/pulls/5330#issue-4038062))

<sub>posted by Sider</sub>

File: mesonbuild/compilers/cpp.py [hunk 5]
```diff
@@ -256,9 +256,9 @@ def language_stdlib_only_link_flags(self):
 
 
 class PGICPPCompiler(PGICompiler, CPPCompiler):
-    def __init__(self, exelist, version, compiler_type, is_cross, exe_wrapper=None, **kwargs):
+    def __init__(self, exelist, version, is_cross, exe_wrapper=None, **kwargs):
         CPPCompiler.__init__(self, exelist, version, is_cross, exe_wrapper, **kwargs)
-        PGICompiler.__init__(self, compiler_type)
+        PGICompiler.__init__(self, CompilerType.PGI_STANDARD)
```
Comment (ignatenkobrain) [inline mesonbuild/compilers/cpp.py]: __[Flake8]__ [F821] undefined name 'CompilerType' ([link](https://sider.review/gh/repos/19784232/pulls/5330#issue-4038063))

<sub>posted by Sider</sub>

**Comment by jpakkane** [comment]
The ICL MR is in. Should this one be just closed now?

**Comment by dcbaker** [comment]
I think so, but I don't have the PGI compilers to test with.

**Comment by jpakkane** [comment]
Those who do have them can file new MRs to fix things. Closing.

**Pull request closed without merging**